HPXMLtoOpenStudio/resources/location.rb

```diff
@@ -40,7 +40,11 @@ def self.apply_site(model, epw_file)
     site.setElevation(epw_file.elevation)
   end
 
-  def self.apply_year(model, hpxml)
+  def self.apply_year(model, hpxml, epw_file)
+    if !epw_file.startDateActualYear.is_initialized && Date.leap?(hpxml.header.sim_calendar_year)
+      fail "Specified a leap year (#{hpxml.header.sim_calendar_year}) with a TMY weather file. A non-leap year, or an AMY weather file with the same actual year, must be specified."
+    end
```

Comment:
I tend to feel that this check is too specific. Yes, a TMY file should have 8760 hours, but an AMY file that someone developed could also have 8760 hours.

How would you feel about this? (Completely untested.)

```suggestion
    n_hours = epw_file.data.size
    if n_hours != 8784 && Date.leap?(hpxml.header.sim_calendar_year)
      fail "Specified a leap year (#{hpxml.header.sim_calendar_year}) but weather data has #{n_hours} hours."
    end
```